### PR TITLE
tools, media: add DM overview

### PR DESCRIPTION
**Pull request by lilin90**
@GregoryIan @csuzhangxc @CaitinChen PTAL
Via: https://github.com/pingcap/docs/blob/master/tools/dm/overview.md

**Note:** The original links in this overview will be added after other related documents are ready. We can do that in another PR later.

### CODE REVIEW

File: tools/dm/overview.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+下面简单介绍 DM 数据同步功能的五大特性。
+
+### 库表路由
+
+库表路由是指将上游 MySQL 或 MariaDB 实例的某些表同步到下游指定表的路由功能，可以用于分库分表的合并同步。
+
+### 库表同步黑白名单
+
+库表同步黑白名单是指上游数据库实例表的黑白名名单过滤规则。其过滤规则类似于 MySQL `replication-rules-db`/`replication-rules-table`，可以用来过滤或只同步某些数据库或某些表的所有操作。
```
Comment (csuzhangxc) [inline tools/dm/overview.md]:
```suggestion
库表同步黑白名单是指上游数据库实例表的黑白名单过滤规则。其过滤规则类似于 MySQL `replication-rules-db`/`replication-rules-table`，可以用来过滤或只同步某些数据库或某些表的所有操作。
```

File: tools/dm/overview.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+### Binlog event 过滤
+
+Binlog event 过滤是比库表同步黑白名单更加细粒度的过滤规则，可以使用 `INSERT`、`TRUNCATE TABLE` 等语句指定只同步或过滤掉某些 `schema/table` 的指定类型的 binlog events。
```
Comment (csuzhangxc) [inline tools/dm/overview.md]: how about 
```
可以指定只同步或者过滤掉某些 `schema/table` 的指定类型的 binlog events， 比如 `INSERT`，`TRUNCATE TABLE`
```

File: tools/dm/overview.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+### 列值转换
+
+列值转换是指根据用户指定的内置表达式对表的列进行转换，可以用来解决分库分表自增主键 ID 的冲突。
```
Comment (csuzhangxc) [inline tools/dm/overview.md]:
```suggestion
列值转换是指根据用户指定的内置表达式对表的列进行转换，可以用来解决分库分表合并时自增主键 ID 的冲突。
```

File: tools/dm/overview.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
++ DDL 语法
+
+    - 目前，TiDB 部分兼容 MySQL 支持的 DDL 语句。因为 DM 使用 TiDB parser 来解析处理 DDL 语句，所以目前仅支持 TiDB parser 支持的 DDL 语法。详见 [TiDB DDL 语法支持](../../sql/ddl.md).
+
+    - DM 遇到不兼容的 DDL 语句时会报错。要解决此报错，需要使用 dmctl 手动处理，要么跳过该 DDL 语句，要么用指定的 DDL 语句来替换它。
+
++ 分库分表
+
+    如果业务分库分表之间存在数据冲突，冲突的列**只有自增主键列**，并且**列的类型是 bigint**，可以尝试使用列值转换来解决；否则不支持同步，有冲突的数据会相互覆盖造成数据丢失。
```
Comment (csuzhangxc) [inline tools/dm/overview.md]:
```suggestion
    如果业务分库分表之间存在数据冲突，冲突的列**只有自增主键列**，并且**列的类型是 bigint**，可以尝试使用列值转换来解决；否则不推荐使用 DM 进行同步，如果进行同步则有冲突的数据会相互覆盖造成数据丢失。
```

File: tools/dm/overview.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
++ 操作限制
+
+    - DM-worker 重启后不能自动恢复数据同步任务，需要使用 dmctl 手动执行 `start-task`。
+    - 在一些情况下，DM-worker 或 dm-master 重启后不能自动恢复 DDL lock 同步，需要手动处理。
```
Comment (csuzhangxc) [inline tools/dm/overview.md]:
```suggestion
    - 在一些情况下，DM-worker 重启后不能自动恢复 DDL lock 同步，需要手动处理。
```

**Comment by lilin90** [comment]
@csuzhangxc Thanks a lot! Addressed comments, PTAL again~

**Comment by csuzhangxc** [comment]
LGTM

**Comment by lilin90** [comment]
@CaitinChen PTAL~ Thanks!

**Pull request merged**